### PR TITLE
add pdfrw and pagelabels

### PR DESCRIPTION
**Pull request by teto**
###### Motivation for this change
I want to package https://github.com/dsanson/termpdf.py (I have but there is an issue when importing fitz so termpdf.py update will be delayed until I fix it).

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by teto** [comment]
@GrahamcOfBorg build pythonPackages.pdfrw pythonPackages.pagelabels

**Comment by teto** [comment]
@GrahamcOfBorg build pythonPackages.termpdfpy

**Comment by teto** [comment]
@GrahamcOfBorg build termpdfpy

**Comment by teto** [comment]
@GrahamcOfBorg build termpdfpy

**Comment by teto** [comment]
@GrahamcOfBorg build termpdfpy

**Comment by teto** [comment]
@GrahamcOfBorg build termpdfpy

**Comment by teto** [comment]
@GrahamcOfBorg eval

**Comment by teto** [comment]
you look a bit bored @jonringer, don't you ? don't worry I have thought about you and prepared this PR just in case :p

File: pkgs/development/python-modules/pagelabels/default.nix [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{ stdenv, buildPythonPackage, fetchPypi, pdfrw }:
```
Comment (jonringer) [inline pkgs/development/python-modules/pagelabels/default.nix]: add a line please :)
```suggestion
{ stdenv, buildPythonPackage, fetchPypi, pdfrw }:

```

File: pkgs/development/python-modules/pdfrw/default.nix [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{ stdenv, buildPythonPackage, fetchPypi, mupdf, swig }:
+buildPythonPackage rec {
```
Comment (jonringer) [inline pkgs/development/python-modules/pdfrw/default.nix]: add a line here as well :)
```suggestion

buildPythonPackage rec {
```

File: pkgs/development/python-modules/pdfrw/default.nix [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  pname = "pdfrw";
+  version = "0.4";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1x1yp63lg3jxpg9igw8lh5rc51q353ifsa1bailb4qb51r54kh0d";
+  };
+
+  # no python3 support yet
+  doCheck = false;
```
Comment (jonringer) [inline pkgs/development/python-modules/pdfrw/default.nix]: if the comment is true, then this should work
```suggestion
  doCheck = isPy27;
```

File: pkgs/development/python-modules/pagelabels/default.nix (hunk 1 above)
Comment (teto) [reply pkgs/development/python-modules/pagelabels/default.nix]: just curious, do you refer to a specific guideline to suggest the added line, or is that sthg you have seen in a formatter ?

**Comment by jonringer** [comment]
isPy27 needs to be added to the top of  the file, and please squash the commits :)

File: pkgs/development/python-modules/pdfrw/default.nix (hunk 3 above)
Comment (teto) [reply pkgs/development/python-modules/pdfrw/default.nix]: turns out that there are even more errors with python2 so I just disabled for both as originally ( I had not tested python2 at the time)

**Comment by jonringer** [comment]
@GrahamcOfBorg build python27Packages.pagelabels python27Packages.pdfrw python37Packages.pagelabels python37Packages.pdfrw termpdfpy

**Comment by jonringer** [comment]
not sure if this intended behavior:
```
[nix-shell:~/.cache/nix-review/pr-69637]$ termpdf.py --help
Terminal does not support reporting screen sizes via the TIOCGWINSZ ioctl
```

**Comment by teto** [comment]
It is, it only works with the kitty terminal for now but is pretty impressive nonetheless (can select and youank from a pdf in a terminal - if it's kitty)

File: pkgs/applications/misc/termpdf.py/default.nix [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    rev = "70fa118080f44d06416f95cb309a0985ee7c457b";
+    sha256 = "dMfvXsUIB8v7pwP1HIDJ/0SxDHwHa8TdraRtew+PQoM=";
+  };
+
+  propagatedBuildInputs = [
+    bibtool pybtex pymupdf pyperclip roman pagelabels pdfrw
+    pynvim setuptools
+  ];
+
+  doCheck = false;
```
Comment (jonringer) [inline pkgs/applications/misc/termpdf.py/default.nix]:
```suggestion
  # upstream doesn't contain tests
  doCheck = false;
```

File: pkgs/applications/misc/termpdf.py/default.nix [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+, bibtool
+, pybtex
+, pymupdf
+, pynvim
+, pyperclip
+, roman
+, pdfrw
+, pagelabels
+, setuptools
+}:
```
Comment (jonringer) [inline pkgs/applications/misc/termpdf.py/default.nix]:
```suggestion
}:

```

File: pkgs/development/python-modules/pagelabels/default.nix [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  pname = "pagelabels";
+  version = "1.0.2";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0xvzr8gz1mripwdjdlfdv58j3mhv23vpqz3d39nj1cg0dic34aca";
+  };
+
+  buildInputs = [ pdfrw ];
+
```
Comment (jonringer) [inline pkgs/development/python-modules/pagelabels/default.nix]:
```suggestion

  # upstream doesn't contain tests
  doCheck = false;
```

File: pkgs/development/python-modules/pdfrw/default.nix (hunk 3 above)
Comment (jonringer) [reply pkgs/development/python-modules/pdfrw/default.nix]: there should be an attempt to run all the pure tests that you can. There will be some that you can't do in a sandbox environment, but we want to guarentee that the package wont break when your dependencies get bumped.

Comment (teto) [reply pkgs/development/python-modules/pdfrw/default.nix:13]: it requires additionnal pdfs https://github.com/pmaupin/pdfrw#6testing . I think it's fine to disable them. The decision can be revisited when and if the package breaks.

File: pkgs/development/python-modules/pagelabels/default.nix [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+buildPythonPackage rec {
+  pname = "pagelabels";
+  version = "1.2.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "07as5kzyvj66bfgvx8bph8gkyj6cgm4lhgxwb78bpdl4m8y8kpma";
+  };
+
+  buildInputs = [ pdfrw ];
```
[the remark this answers is not in the document]
Comment (doronbehar) [reply pkgs/development/python-modules/pagelabels/default.nix]: Fix is in:

- https://github.com/NixOS/nixpkgs/pull/477554

**Pull request merged**